src/mame/sinclair/sprinter.cpp

```diff
@@ -0,0 +1,1751 @@
+	case 0xcb:
+		m_hold = data;
+		break;
+	case 0xc4:
+	case 0xcc:
+		m_port_y = data;
+		break;
+	case 0xc5:
+	case 0xcd:
+		m_screen->update_now();
+		m_rgmod = data;
+		update_int(true);
+		break;
+
+	case 0xc6: // CNF/SYS
+	case 0xce:
+		m_ram_sys = BIT(~offset, 6);
+		if (BIT(data, 1))
+		{
+			m_turbo = BIT(data, 0);
+			update_cpu();
+		}
+		else
+			m_arom16 = BIT(data, 0);
+
+		if (BIT(data, 2))
+		{
+			m_cnf = data;
+			if (BIT(m_cnf, 5)) m_pn &= 0xc0;  // CNF_PN[5..0]_RESET
+			if (BIT(m_cnf, 6)) m_sc = 0;      // CNF_SC_RESET
+			if (BIT(~m_cnf, 7)) m_pn &= 0x1f; // CNF_PN[7..6]_RESET
+		}
+
+		update_memory();
+		break;
+
+	case 0xc7:
+	case 0xcf:
+		m_alt_acc = 1;
+		m_aagr = (offset & 0x300) | data;
+		m_xcnt = BIT(offset, 10, 6);
+		m_xagr = 0;
+		break;
+
+	case 0xd0 ... 0xef:
+		update_memory();
+		break;
+	case 0xf0 ... 0xff:
+		m_ram_pages[m_pg3] = data;
+		update_memory();
+		break;
+
+	// no port
+	case 0x00:
+		break;
+
+	default:
+		LOGIO("IOw: %02X / %X < %x\n", dcpp, offset, data);
+		break;
+	}
+}
+
+void sprinter_state::accel_control_r(u8 data)
+{
+	const bool is_prefix = (data == 0xcb) || (data == 0xdd) || (data == 0xed) || (data == 0xfd);
+	if (!is_prefix && !m_prf_d) // neither prefix nor prefixed
+	{
+		if ((((data & 0x1b) == 0x00) || ((data & 0x1b) == 0x09) || ((data & 0x1b) == 0x12) || ((data & 0x1b) == 0x1b))
+			&& (((data & 0xe4) == 0x40) || ((data & 0xe4) == 0x64)))
+		{
+			m_acc_dir = ((data & 7) == OFF_HALT) ? OFF : static_cast<accel_state>(data & 7);
+			m_fn_acc = MODE_NOP;
+		}
+		else {
+			const accel_state state_candidate = static_cast<accel_state>(data);
+			switch(state_candidate)
+			{
+			case MODE_AND:
+			case MODE_XOR:
+			case MODE_OR:
+				m_fn_acc = state_candidate;
+				break;
+			default:
+				break;
+			}
+		}
+	}
+	m_prf_d = is_prefix;
+}
+
+void sprinter_state::accel_r_tap(offs_t offset, u8 &data)
+{
+	const std::string m = (m_fn_acc == MODE_AND) ? "&" : (m_fn_acc == MODE_OR) ? "|" : (m_fn_acc == MODE_XOR) ? "^" : "";
+	if (m_acc_dir == SET_BUFFER)
+	{
+		m_acc_cnt = data;
+		LOGACCEL("Accel buffer: %d\n", m_acc_cnt);
+	}
+	else if (m_pages[offset >> 14] & BANK_RAM_MASK) // block ops RAM only
+	{
+		const u16 acc_cnt = m_acc_cnt ? m_acc_cnt : 256;
+		if (m_acc_dir == COPY)
+		{
+			LOGACCEL("Accel rCOPY: %s%02x\n", m, offset);
+			for (auto i = 0; i < acc_cnt; i++)
+			{
+				if (i && !machine().side_effects_disabled())
+					do_cpu_wait();
+				const u16 addr = offset + i;
+				data = (m_pages[addr >> 14] & BANK_RAM_MASK) ? ram_r(addr) : 0xff;
+				update_accel_buffer(i, data);
+			}
+		}
+		else if (m_acc_dir == COPY_VERT)
+		{
+			LOGACCEL("Accel rCOPY_GR: %s%02x (%x)\n", m, offset, m_port_y);
+			for (auto i = 0; i < acc_cnt; i++)
+			{
+				if (i && !machine().side_effects_disabled())
+					do_cpu_wait();
+				data = ram_r(offset);
+				update_accel_buffer(i, data);
+				m_port_y++;
+			}
+		}
+		else if (m_acc_dir == FILL_VERT)
+			m_port_y += acc_cnt;
+		else
+			return;
+	}
+}
```

Comment:
Does the final `else return;` serve a purpose?  It will fall off the end of the function anyway.